dipy/tracking/vox2track.pyx

```diff
@@ -162,6 +161,176 @@ def streamline_mapping(streamlines, voxel_size=None, affine=None,
+     edge[1] = floor(p_y + eps) if d_y >= 0.0 else ceil(p_y - eps)
+     edge[2] = floor(p_z + eps) if d_z >= 0.0 else ceil(p_z - eps)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+def _streamlines_in_mask(list streamlines,
+                         cnp.uint8_t[:,:,:] mask,
+                         lin_T, offset):
+    """Filters streamlines based on whether or not they pass through a ROI,
+    using a line-based algorithm for compressed streamlines.
+
+    This function is private because it's supposed to be called only by
+    tracking.utils.target_line_based.
+
+    Parameters
+    ----------
+    streamlines : sequence
+        A sequence of streamlines.
+    mask : array-like (uint8)
+        A mask used as a target. Non-zero values are considered to be within
+        the target region.
+    lin_T : array (3, 3)
+        Transpose of the linear part of the mapping to voxel space. Obtained
+        with `_mapping_to_voxel`.
+    offset : array or scalar
+        Mapping to voxel space. Obtained with `_mapping_to_voxel`.
+
+    Returns
+    -------
+    in_mask : 1D array of bool (uint8), one for each streamline.
+        0 if passing through mask, 1 otherwise
+        (2 for single-point streamline)
+    """
+    cdef cnp.double_t[:,:] voxel_indices
+
+    # Declaring it here and passing to friend function because creating them
+    # there would cost too much.
+    cdef cnp.double_t[:] current_pt = np.zeros(3, dtype=np.double)
+    cdef cnp.double_t[:] next_pt = np.zeros(3, dtype=np.double)
+    cdef cnp.double_t[:] direction = np.zeros(3, dtype=np.double)
+    cdef cnp.double_t[:] current_edge = np.zeros(3, dtype=np.double)
+
+    cdef int nb_streamlines = len(streamlines)
+    cdef cnp.uint8_t[:] in_mask = np.zeros(nb_streamlines, dtype=np.uint8)
+    cdef int streamline_idx
+
+    for streamline_idx in range(nb_streamlines):
+        # Can't call _to_voxel_coordinates because it casts to int
+        voxel_indices = np.dot(streamlines[streamline_idx], lin_T) + offset
+
+        in_mask[streamline_idx] = _streamline_in_mask(
+            voxel_indices, mask, current_pt, next_pt, direction, current_edge)
+
+    return np.asarray(in_mask)
+
+
+@cython.boundscheck(False)
+@cython.wraparound(False)
+@cython.cdivision(True)
+cdef int _streamline_in_mask(cnp.double_t[:,:] streamline,
+                             cnp.uint8_t[:,:,:] mask,
+                             cnp.double_t[:] current_pt,
+                             cnp.double_t[:] next_pt,
+                             cnp.double_t[:] direction,
+                             cnp.double_t[:] current_edge) nogil:
+    """
+    Check if a single streamline is passing through a mask. This ia an utility
+    function to make streamlines_in_mask() more readable.
+    """
+    cdef cnp.double_t direction_norm, remaining_distance
+    cdef cnp.double_t length_ratio, half_ratio
+    cdef int point_idx, dim_idx
```

Comment:
Rather than int use ```cnp.npy_intp``` variable that makes sure you will be using the correct integer data types in different platforms. Change everywhere. It's good practice.